python/src/aiconfig/editor/client/src/components/EditorContainer.tsx

```diff
@@ -626,6 +633,16 @@ export default function EditorContainer({
   );
 
   const isDirty = aiconfigState._ui.isDirty !== false;
+  useEffect(() => {
+    if (!isDirty) {
+      return;
+    }
+
+    // Save every 15 seconds if there are unsaved changes
+    const saveInterval = setInterval(onSave, AUTOSAVE_INTERVAL_MS);
+
+    return () => clearInterval(saveInterval);
+  }, [isDirty, onSave]);
```

Comment:
Does `isDirty` being changed retrigger an instant save?

Reply:
No, it happens at the end of the interval, so 15s after becoming dirty